### ISSUE TITLE
Refactor chat model, add support for custom system prompts, maybe logit masks

### ISSUE THREAD

**Issue by tarasglek** [tarasglek/chatcraft.org#104]
This isn't super actionable, but something to keep an eye on for reference.

https://github.com/minimaxir/simpleaichat/ is so far the coolest api i've seen to openai. Main observations there are that langchain is not chat-first and that messing with system prompts is super powerful.

We should look to match some of these features.

System-prompt via markdown bulletpoints(can make it easy to add/remove various tuning), being more agressive re avoiding convos by using stop token): https://github.com/minimaxir/simpleaichat/blob/main/examples/notebooks/simpleaichat_coding.ipynb

How to use logits and completely restrict openai output(useful for classification): https://github.com/minimaxir/simpleaichat/blob/main/examples/notebooks/chatgpt_inline_tips.ipynb

**Comment by humphd** [comment]
Now that messages are editable, being able to show and edit the system prompt would be really useful.  For the most part, I think users won't need/want to see the system prompt, so we should consider what the UX is for revealing it in your current chat so you can play with it.  Do we put some button or something to "Show System Prompt" in the chat header?

<img width="1170" alt="Screenshot 2023-06-13 at 4 26 34 PM" src="https://github.com/tarasglek/chatcraft.org/assets/427398/025fc56a-70f6-4b47-89ce-b6e787335804">

I don't know enough about what "logit masks" are to understand the rest of this.

**Comment by humphd** [comment]
This is fixed and shipped to prod.  I don't know what you mean by "How to use logits and completely restrict openai output(useful for classification): https://github.com/minimaxir/simpleaichat/blob/main/examples/notebooks/chatgpt_inline_tips.ipynb", so can you file a new issue on that if you want us to do something with it?
